### ISSUE TITLE
ENH: Support using repr() instead of str() to print non-numeric Series elements, especially strings

### ISSUE THREAD

**Issue by gwerbin** [pandas-dev/pandas#46744]
#### Is your feature request related to a problem?

Sometimes, it is useful for debugging to be able to see the contents of a Series with `repr()` instead of `str()` applied to each element.

Specifically with strings, it is often useful to see non-ASCII Unicode characters escaped, and to see the surrouning quotation marks, to detect whitespace at the start and end.

For example, `print(pd.Series(['a', 'b', 'c'], dtype='string'))` yields:
```
0    a
1    b
2    c
dtype: string
```

But it would be helpful if it could yield:
```
0    'a'
1    'b'
2    'c'
dtype: string
```

#### Describe the solution you'd like

One possibility is a new module option, maybe `display.string_as_repr`, to enable this behavior.

#### API breaking implications

None that I know of.

#### Describe alternatives you've considered

You could use `.apply(repr)` or similar, to see this output in an ad-hoc way.

#### Additional context

This was originally a StackOverflow question: https://stackoverflow.com/q/71832297/2954547

**Comment by rhshadrach** [comment]
I wonder what the downsides would be in making this default behavior.

```
ser = pd.Series(['  a', 'a', ' a'])
print(ser)
print(ser.apply(repr))

0      a
1      a
2      a
dtype: object
0    '  a'
1      'a'
2     ' a'
dtype: object
```

I could see trying to perform logic while not realizing there are space characters giving rise to issues.

**Comment by gwerbin** [comment]
Something else I thought of: perhaps this could also be added as a parameter to `DataFrame.to_string`? Not sure what the core devs' opinions are on this kind of thing.

**Comment by MarcoGorelli** [comment]
I'm not keen on adding an extra parameter (`.apply(repr)` should work fine), but changing the default sounds nice - I've added that to the agenda for this month's dev meeting

**Comment by rhshadrach** [comment]
Agreed on not adding a parameter. Looking at this again, I think using repr as the default would be confusing. When one does `print(obj)`, the `__str__` method, not `__repr__`  is called. It seems like we'd be going against this standard.

**Comment by EkinsMatthew** [comment]
Hi all, wondering where we are on this. I see from the related post #50194 it was brought up at the Dev meeting. I too would really appreciate __repr__ being the default display method for strings.

[Here is a link](https://stackoverflow.com/q/76426485/13261310) to a stackoverflow post I made a yesterday showing a good example of how this makes analysis on string columns with escaped characters very difficult.

**Comment by rhshadrach** [comment]
It does seem like Python and NumPy both use repr in containers consistently.

```
class Foo:
    def __str__(self):
        return 'str'

    def __repr__(self):
        return 'repr'

foo = Foo()
print(foo)
# str
print([foo])
# [repr]
print((foo,))
# (repr,)
print({foo: foo})
# {repr: repr}
print(np.array([foo]))
# [repr]
print(pd.Series([foo]))
# 0    str
# dtype: object
```

This suggests to me pandas should do the same, but I wonder what issues (especially for downstream libraries) this might cause.

**Comment by mroeschke** [comment]
I think this is a sensible change. Not sure how easy it is to migrate from str to repr though. Maybe it's a candidate for a 3.0 breaking change?

**Comment by rhshadrach** [comment]
The only other way I see doing this is via an option. +1 on making this a breaking change.
